microsoft/testsuites/core/storage.py

```diff
@@ -270,6 +282,44 @@ def verify_scsi_disk_controller_type(self, node: RemoteNode) -> None:
     def verify_nvme_disk_controller_type(self, node: RemoteNode) -> None:
         self._verify_disk_controller_type(node, DiskControllerType.NVME)
 
+    @TestCaseMetadata(
+        description="""
+        This test verifies the nvme drivers are either built in or present in initramfs.
+        This test can be used to verify an image before it is tagged as NVMe capable.
+        """,
+        priority=1,
+    )
+    def verify_supports_nvme(self, node: RemoteNode, environment: Environment) -> None:
+        _has_nvme_core = node.tools[KernelConfig].is_built_in("CONFIG_NVME_CORE") or (
+            node.tools[KernelConfig].is_built_as_module("CONFIG_NVME_CORE")
+            and node.tools[Lsinitrd].has_module("nvme-core.ko")
+        )
+        _has_nvme = node.tools[KernelConfig].is_built_in("CONFIG_BLK_DEV_NVME") or (
+            node.tools[KernelConfig].is_built_as_module("CONFIG_BLK_DEV_NVME")
+            and node.tools[Lsinitrd].has_module("nvme.ko")
+        )
+
+        # Pass condition
+        if _has_nvme_core and _has_nvme:
+            return
+
+        if environment.platform:
+            location = node.capability._extended_runbook.location
```

Comment:
It looks need an extra field on disk settings called `supported_disk_controller_type`. So, the checks can be on `node.features[Disk].supported_disk_controller_type`.

Reply:
LISA doesn't currently gather any data about the image features, so I'm not sure the best way to implement this. Thoughts?

Reply:
I feel like there would need to be an image equivalent to `_resource_sku_to_capability` which will allow the image features determine the settings and behaviors of NodeSpace and Features.

Reply:
Ok, I thought it's to get features on vm sizes. Yes, we need to support image info in test case as well. There is an `AzureImageSchema` class in Azure. We can create a common ImageSchema in `NodeSpace` and add needed information in platform.